src/pages/skills/AddSkill.js

```diff
@@ -0,0 +1,77 @@
+import { useCallback, useState } from "react"
+import { useSelector, useDispatch } from "react-redux"
+import { useEffect } from "react"
+import { Link, useNavigate, useParams } from "react-router-dom"
+import { getSkills, reset, postSkill } from "../../features/skills/skillsSlice"
+import DefaultContainer from "../../layout/DefaultContainer"
+import ContentContainer from "../../layout/ContentContainer"
+import Skill from "../../layout/Skill"
+import { Title } from "../../elements/Title"
+import { StyledInput } from "../../elements/Input"
+import { Button } from "../../elements/Button"
+
+function AddSkill() {
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+  const [newSkill, setNewSkill] = useState()
+  const handleAddNewSkill = useCallback(
+    async (event) => {
+      event.preventDefault()
+      setNewSkill(event.target.value)
+    },
+    [setNewSkill]
+  )
+  const onSubmitNewSkill = useCallback(
+    async (event) => {
+      event.preventDefault()
+      const result = await dispatch(postSkill({ skill_name: newSkill }))
+      if (result.payload.status === 200) {
+        dispatch(getSkills())
+      }
+    },
+    [dispatch, setNewSkill, newSkill]
+  )
+  useEffect(() => {
+    const isUser = localStorage.getItem("user_token")
+    if (!isUser) {
+      navigate("/auth/sign-in")
+    }
+    dispatch(getSkills())
+    return () => {
+      dispatch(reset())
+    }
+  }, [dispatch])
+  const { skills } = useSelector((state) => state.skills)
```

Comment:
Let's move skills state to the top. It's the most important data in this component, would be good to have skills state visible right away.